pkg/wireguard/wireguard.go

```diff
@@ -133,24 +136,68 @@ func (dev *Device) Run(ctx context.Context, peers PeerSource) error {
 		case <-device.Wait():
 			return nil
 		case peerEvent := <-peers.EventCh():
-			cfg := wgtypes.Config{
-				Peers: []wgtypes.PeerConfig{
-					{
-						PublicKey:         peerEvent.PubKey,
-						Remove:            peerEvent.Remove,
-						ReplaceAllowedIPs: true,
-						AllowedIPs: []net.IPNet{
-							*netaddr.IPPrefixFrom(peerEvent.Address, peerEvent.Address.BitLen()).IPNet(),
-						},
-					},
-				},
+			if err := dev.handlePeerEvent(client, logger, peerEvent); err != nil {
+				return err
 			}
+		}
+	}
+}
+
+func (dev *Device) handlePeerEvent(client *wgctrl.Client, logger *zap.Logger, peerEvent PeerEvent) error {
+	if !peerEvent.Remove {
+		oldCfg, err := client.Device(interfaceName)
+		if err != nil {
+			return fmt.Errorf("error retrieving Wireguard configuration: %w", err)
+		}
+
+		// check if this update can be skipped
+		pubKey := peerEvent.PubKey.String()
+
+		for _, oldPeer := range oldCfg.Peers {
+			if oldPeer.PublicKey.String() == pubKey {
+				if len(oldPeer.AllowedIPs) != 1 {
```

Comment:
Do fully-instantiated peers ever really only have a single allowed IP?  At minimum, they will have at least one Node IP and at least one Pod subnet, no?  If that is the case, does this optimisation really help anything?

Reply:
Never mind!  This is SiderLink.  They'll _always_ have a single IP.

Reply:
they only have one single allowed IP - the SideroLink address, we don't want to route any other IPs through SideroLink, as it's pure p2p interface.